extensions/smallrye-graphql-client/runtime/src/main/java/io/quarkus/smallrye/graphql/client/runtime/GraphQLClientBuildConfig.java

```diff
@@ -0,0 +1,16 @@
+package io.quarkus.smallrye.graphql.client.runtime;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "smallrye-graphql-client", phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+public class GraphQLClientBuildConfig {
+
+    /**
+     * Configuration item to enable build-time scanning in Quarkus for generating typesafe GraphQL client models.
+     * If true, build-time scanning is enabled. By default, it is true.
+     */
+    @ConfigItem(name = "enable-build-time-scanning", defaultValue = "true")
+    public boolean allowBuildTimeScanning;
```

Comment:
Hmm, maybe rename  the field to `enableBuildTimeScanning` and remove the explicit name above.. 'Allow' sounds like it has something to do with security, which it doesn't..

Reply:
Yeah in general, let's try to be consistent and avoid the `name` attribute when you can.